office365_connector.py

```diff
@@ -486,30 +423,82 @@ def load_state(self):
         :return: loaded state
         """
         state = super().load_state()
-        try:
-            state = _decrypt_state(state, self._asset_id)
-        except Exception as e:
-            self._dump_error_log(e)
-            self.debug_print("{}: {}".format(MSGOFFICE365_DECRYPTION_ERROR, str(e)))
-            state = None
-
-        return state
+        if not isinstance(state, dict):
+            self.debug_print("Resetting the state file with the default format")
+            state = {
+                "app_version": self.get_app_json().get('app_version')
+            }
+            return state
+        return self._decrypt_state(state, self._asset_id)
 
     def save_state(self, state):
         """
-        Encrypt and save the current state dictionary to the the state file.
+        Encrypt and save the current state dictionary to the state file.
 
         :param state: state dictionary
         :return: status
         """
-        try:
-            state = _encrypt_state(state, self._asset_id)
-        except Exception as e:
-            self._dump_error_log(e)
-            self.debug_print("{}: {}".format(MSGOFFICE365_ENCRYPTION_ERROR, str(e)))
-            return phantom.APP_ERROR
+        return super().save_state(self._encrypt_state(state, self._asset_id))
+
+    def update_state_fields(self, state, salt, helper_function, error_message):
```

Comment:
You've applied my suggestion only in half, as each of those 3 blocks below, like 

```
        if state.get("non_admin_auth", {}).get("refresh_token"):
            try:
                state["non_admin_auth"]["refresh_token"] = helper_function(
                    state["non_admin_auth"]["refresh_token"], salt
                )
            except Exception as ex:
                self.debug_print("{}: {}".format(error_message,
                                                              _get_error_msg_from_exception(ex, self)))
                state["non_admin_auth"]["refresh_token"] = None
```

is the same except for the structure updated. You could have a function and simply call it 3 times, passing:
`state.get("non_admin_auth", {}).get("access_token")`, `state.get("non_admin_auth", {}).get("refresh_token")` and `state.get("admin_auth", {}).get("access_token")` to it

Reply:
I have made the changes please check and let me know if you have any question